### ISSUE TITLE
Get Doxygen documentation on Read the Docs

### ISSUE THREAD

**Issue by rgov** [stp/stp#283]
We should configure Read the Docs to generate Doxygen output.

There's a tool, Breathe, [which can be used](https://breathe.readthedocs.io/en/latest/readthedocs.html).

**Comment by msoos** [comment]
But why? I'd say a manpage with help2man would be more important than a Doxygen doc? I have done the help2man with CMS and it's pretty cool actually: https://github.com/msoos/cryptominisat/issues/466

**Comment by rgov** [comment]
It looks like that is for documenting the command-line options. Doxygen is for converting comments in source code into a browsable interface. I use [Clang's Doxygen page](https://clang.llvm.org/doxygen/classclang_1_1ento_1_1SymExpr.html) all the time, I can't imagine how terrible it would be to browse it via `man`.

Not opposed to having manpages as well.

**Comment by msoos** [comment]
Actually, I find that source code docs go out of date super-duper fast so I would be reluctant to do that for CMS -- I did it once, put about a full month of effort into it, and it got outdated within a year. So, if you are happy to do it, I'm all for it, but unless you plan to do it for something very stable, like the API, I think it will get outdated relatively fast :(

OK, I'll do the manpage thing :) Opened issue here: https://github.com/stp/stp/issues/284

**Comment by rgov** [comment]
That's the problem Read the Docs is trying to solve: It has a post commit hook that re-generates the documentation. And it keeps snapshots of tagged releases, so people can refer to the old stuff, or review the new stuff. It just takes the discipline to update comments when behavior changes.

**Comment by msoos** [comment]
Yeah, it's the discipline thing :D But I agree that for the API, this would be absolutely perfect, and much-needed.
